hs_core/views/__init__.py

```diff
@@ -382,6 +382,82 @@ def unshare_resource_with_user(request, shortkey, user_id, *args, **kwargs):
 
     return HttpResponseRedirect(request.META['HTTP_REFERER'])
 
+
+def resource_labeling_action(request, shortkey, *args, **kwargs):
+    """
+    This must be a POST request. The following data needs to be passed as part of the POST request:
+    action: resource labeling action - has to be either 'CREATE' or 'DELETE'
+    label_type: type of label - one of these: 'LABEL', 'FAVORITE', 'MINE', 'FOLDER', 'SAVEDLABEL'
+    label: a string value required if the label_type is either 'LABEL', 'SAVEDLABEL',
+           or 'FOLDER' and action is 'CREATE'
+
+    """
+    _LABEL = 'LABEL'
+    _FOLDER = 'FOLDER'
+    _FAVORITE = 'FAVORITE'
+    _MINE = 'MINE'
+    _SAVEDLABEL = 'SAVEDLABEL'
```

Comment:
No clear reason to prepend these all with `_` it just makes them slightly more obscure to read for no reason.